doc/source/data/glossary.rst

```diff
@@ -0,0 +1,154 @@
+.. _datasets_glossary:
+
+=====================
+Ray Datasets Glossary
+=====================
+
+.. glossary::
+
+    Batch format
+        The way batches of data are represented.
+
+        Set ``batch_format`` in methods like :meth:`~ray.data.Dataset.iter_batches` and
+        :meth:`~ray.data.Dataset.map_batches` to specify the batch type.
+
+        .. doctest::
+
+            >>> import ray
+            >>> dataset = ray.data.range_table(10)
+            >>> next(iter(dataset.iter_batches(batch_format="numpy", batch_size=5)))
+            {'value': array([0, 1, 2, 3, 4])}
+            >>> next(iter(dataset.iter_batches(batch_format="pandas", batch_size=5)))
+               value
+            0      0
+            1      1
+            2      2
+            3      3
+            4      4
+
+        To learn more about batch formats, read
+        :ref:`UDF Input Batch Formats <transform_datasets_batch_formats>`.
+
+    Block
+        A processing unit of data. A :class:`~ray.data.Dataset` consists of a list of
```

Comment:
Nit: "list" implies ordering and we're starting to move away from preserving order, so I think that "collection" would be slightly better.
```suggestion
        A processing unit of data. A :class:`~ray.data.Dataset` consists of a collection of
```